rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -56,31 +58,53 @@ SequentialWriter::SequentialWriter(
   storage_(nullptr),
   metadata_io_(std::move(metadata_io)),
   converter_(nullptr),
+  compressor_{nullptr},
   max_bagfile_size_(rosbag2_storage::storage_interfaces::MAX_BAGFILE_SIZE_NO_SPLIT),
   topics_names_to_info_(),
-  metadata_()
-{}
+  metadata_(),
+  compression_mode_{CompressionMode::NONE} {}
+
 
 SequentialWriter::~SequentialWriter()
 {
   reset();
 }
 
-void SequentialWriter::init_metadata()
+void SequentialWriter::init_compression(const CompressionOptions & compression_options)
```

Comment:
I don't understand why this method isn't part of init_metadata: it's a bit confusing in ```SequentialWriter::open``` why you're calling two init functions taking the same input. I'd recommend the following:

```
SequentialWriter::open(...) 
{
    ....
    init_metadata(compression_options);
    ....
}

SequentialWriter::init_metadata(const CompressionOptions & compression_options);
{
    ....
    init_metadata(...);
    ....
}

SequentialWriter:: init_compression(const CompressionOptions & compression_options);
{
    ....
}

```

I'd also rename the method ```init_compression``` to ```init_compressor```.

If you don't want to combine methods, consider initializing your metadata first before creating the compressor_ resource.